### PR TITLE
Focus-appearance - Idea for bounding the content

### PR DESCRIPTION
**Pull request by alastc**
[Preview](https://raw.githack.com/w3c/wcag/focus-appearance-content-term/understanding/22/focus-appearance-minimum.html) (if rawgit is working).

### CODE REVIEW

**Comment by patrickhlauke** [comment]
On the face of it, looks good, but based on our recent discussions (element CSS border-box vs visual vs "what a user may think is a reasonable representation of the control") the minimum bounding box definition may need further tweaking/expanding/nuance

**Comment by alastc** [comment]
Yep, this is basically a backup idea if the ones in the survey / comments don't work.

The [min-bounding box definition](https://www.w3.org/TR/WCAG22/#dfn-minimum-bounding-box) probably needs some clarification that the 'content' could be text or non-text, and doesn't include things like backgrounds / borders / box-shadows.

**Comment by patrickhlauke** [comment]
> doesn't include things like backgrounds / borders / box-shadows

even box-shadows are a funny one. while yes, some sites use CSS box shadows to effectively make an outline, other sites might use a very subtle, smudged, drop-shadow-y shadow. it's arguably if a real-world user when asked would say that the extent of the drop shadow is also part of the component/element, or if it's "just a shadow behind it".

**Comment by detlevhfischer** [comment]
This could work. I guess with all word-smithing we will not get out of real-world situations where measurement will be difficult (think of icons as content of buttons that are somewhat larger). I wonder whether there good ways to calculate bounding boxes around content, especially if you effectively need a 3:1 cut-off rule for visual components that include gradients / differently colored areas.

**Comment by patrickhlauke** [comment]
> I wonder whether there are established ways of automatically calculating bounding boxes around content

i fear that any automated way of doing it would make certain assumptions that then can lead to false positives/negatives. and it will again depend on what exactly the bounding box is calculated automatically on (the actual thing that receives focus, or some deeper nested container in that ... as thrashed out in far too much back-and-forth in #2226 )

**Comment by alastc** [comment]
> think of icons as content of buttons that are somewhat larger

I think what we should stress in the understanding document is that, despite the exact measure, the best way of meeting it is an outline. It makes it easier to calculate that an outline passes because there shouldn't be any niche cases there. 

It is only if you shrink beneath the content of the control or use an odd shape you'd be calculating it.

> I wonder whether there are established ways of automatically calculating bounding boxes around content

Possibly, but you'd probably have to take a warning-only approach rather than definite fail. Conversely, it would be easy to find certain types of pass, e.g. 2px outline that contrasts with the unfocused state.

File: understanding/22/focus-appearance-minimum.html [hunk 1]
```diff
@@ -20,20 +20,20 @@ <h2>Focus Appearance (Minimum) Success Criterion text</h2>
...
         
-            <p>When <a>user interface components</a> receive keyboard focus, an area of the focus indicator meets the following:</p>
+            <p>When a <a>user interface component</a> has keyboard focus, the focus indicator meets all the following:</p>
             <ul>
-                <li><strong>Minimum area:</strong> The area is either:
+                <li><strong>Contrasting area:</strong> The focus indicator has an area with a contrast ratio of at least 3:1 between the colors when the component is focused and when it is not focused, and that area is at least as large as either:
                 <ul>
-                    <li>at least as large as the area of a 1 <a>CSS pixel</a> thick <a>perimeter</a> of the unfocused component, or</li>
-                    <li>at least as large as the area of a 4 CSS pixel thick line along the shortest side of a <a>minimum bounding box</a> of the unfocused component, and no part of the area is thinner than 2 CSS pixels.</li>
+                    <li>the area of a 1 <a>CSS pixel</a> thick <a>minimum bounding box</a> around the content of the component, or</li>
```
Comment (36degrees) [inline understanding/22/focus-appearance-minimum.html:27]: Minor thing – perimeter is still a defined term but no longer appears in the success criteria.

**Pull request closed without merging**